addons/controllers/csi/vspherecsiconfig_utils.go

```diff
@@ -172,39 +171,8 @@ func (r *VSphereCSIConfigReconciler) mapCSIConfigToProviderServiceAccount(vspher
 			Namespace: vsphereCluster.Namespace,
 		},
 		Spec: capvvmwarev1beta1.ProviderServiceAccountSpec{
-			Ref: &v1.ObjectReference{Name: vsphereCluster.Name, Namespace: vsphereCluster.Namespace},
-			Rules: []rbacv1.PolicyRule{
-				{
-					APIGroups: []string{"vmoperator.vmware.com"},
-					Resources: []string{"virtualmachines"},
-					Verbs:     []string{"get", "list", "watch", "update", "patch"},
-				},
-				{
-					APIGroups: []string{"cns.vmware.com"},
-					Resources: []string{"cnsvolumemetadatas", "cnsfileaccessconfigs"},
-					Verbs:     []string{"get", "list", "watch", "update", "create", "delete"},
-				},
-				{
-					APIGroups: []string{"cns.vmware.com"},
-					Resources: []string{"cnscsisvfeaturestates"},
-					Verbs:     []string{"get", "list", "watch"},
-				},
-				{
-					APIGroups: []string{""},
-					Resources: []string{"persistentvolumeclaims"},
-					Verbs:     []string{"get", "list", "watch", "update", "create", "delete"},
-				},
-				{
-					APIGroups: []string{""},
-					Resources: []string{"persistentvolumeclaims/status"},
-					Verbs:     []string{"get", "update", "patch"},
-				},
-				{
-					APIGroups: []string{""},
-					Resources: []string{"events"},
-					Verbs:     []string{"list"},
-				},
-			},
+			Ref:              &v1.ObjectReference{Name: vsphereCluster.Name, Namespace: vsphereCluster.Namespace},
+			Rules:            providerServiceAccountRBACRules,
 			TargetNamespace:  "vmware-system-csi",
```

Comment:
question(non-blocking): This is orthogonal, but I guess using a constant variable is better than hardcoding. We could follow up on a separate transaction.

Reply:
yup the csi controller needs some refactoring